### PR TITLE
Add logging for normal Bind errors

### PR DESCRIPTION
**Pull request by jpeeler**
This simply differentiates a broker failure response as outlined in the
spec versus some other unknown error.

### CODE REVIEW

**Comment by MHBauer** [comment]
Needs test.

**Comment by jpeeler** [comment]
It looks like at present an unbind request should also be sent in the unexpected error case, but I know that orphan handling is something that is in review now and won't be forgotten.

File: pkg/controller/controller_binding.go [hunk 1]
```diff
@@ -221,6 +221,30 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) error {
...
+				httpErr := err.(osb.HTTPStatusCodeError)
+				s := fmt.Sprintf("Error creating Binding \"%s/%s\" for Instance \"%s/%s\" of ServiceClass %q at Broker %q with status code %d: ErrorMessage: %v, Description: %v",
+					binding.Name,
+					binding.Namespace,
+					instance.Namespace,
+					instance.Name,
+					serviceClass.Name,
+					brokerName,
+					httpErr.StatusCode,
+					*httpErr.ErrorMessage,
```
Comment (pmorie) [inline pkg/controller/controller_binding.go]: Use the `%v` format string for string pointers - it's nil safe, dereferencing the pointers like you have done here is not.

Comment (jpeeler) [inline pkg/controller/controller_binding.go]: That's true, but seeing a pointer address isn't all that helpful. See if you like https://github.com/pmorie/go-open-service-broker-client/pull/44 and I can adjust to use the error method instead.

Comment (pmorie) [inline pkg/controller/controller_binding.go]: Nevermind, it's nil safe but doesn't give us the behavior we want.  You'll need to make variables that hold the value to be printed ('<nil>' if not set), and print those.  Dereferencing these pointers is not safe.

**Comment by jpeeler** [comment]
This now depends on PR https://github.com/kubernetes-incubator/service-catalog/pull/1000

**Pull request merged**